Task_2_3_1

andrey-dru-me1 opened the pull request.

andrey-dru-me1 committed:
implement snake model

andrey-dru-me1 committed:
add field implementation

andrey-dru-me1 committed:
fix test

andrey-dru-me1 committed:
improve apple generating;
add point class;

andrey-dru-me1 committed:
add exceptions

andrey-dru-me1 committed:
add snake in snake exception test

andrey-dru-me1 committed:
change some logic connected to apple

andrey-dru-me1 committed:
add apple test

andrey-dru-me1 committed:
separate tests

andrey-dru-me1 committed:
add data saver

andrey-dru-me1 committed:
add data loader

andrey-dru-me1 committed:
add move() method with direction parameter

andrey-dru-me1 committed:
add game class and its test

andrey-dru-me1 committed:
change structure a little;
add gui window without any info yet

andrey-dru-me1 committed:
add first movements for implementing gui

andrey-dru-me1 committed:
implement graphical interface for game. Doesn't show apples

andrey-dru-me1 committed:
add defeat screen

andrey-dru-me1 committed:
Merge branch 'main' into task-2-3-1

andrey-dru-me1 committed:
add victory screen

andrey-dru-me1 committed:
add application stopping

andrey-dru-me1 committed:
add possibility to start application from main method

andrey-dru-me1 committed:
fix module-info.java so both application and tests runs correctly

andrey-dru-me1 committed:
implement correct work finalisation with timer thread interrupting

andrey-dru-me1 committed:
divide main class to presenters and SnakeEntry

andrey-dru-me1 committed:
modify resource folder access so there is no troubles connected with gradlew run launch

andrey-dru-me1 committed:
reformat code

andrey-dru-me1 committed:
reformat some code

andrey-dru-me1 committed:
create one cell class with list of objects on it (for an extensibility)

andrey-dru-me1 committed:
reformat some files a little

andrey-dru-me1 committed:
make window resizeable

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/SnakeEntry.java:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  private Scene initScene() {
+    scene = new Scene(grid);
```
FXML

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/model/Game.java:
```diff
@@ -0,0 +1,58 @@
+    return crash;
+  }
+
+  public void start() {
```
Я вижу действия, это точно не модель

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/model/point/Point.java:
```diff
@@ -0,0 +1,3 @@
+package ru.nsu.fit.oop.melnikov.game.snake.model.point;
+
+public record Point(int x, int y) { }
```
Можно использовать Pair

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/model/field/cell/objects/CellObject.java:
```diff
@@ -0,0 +1,8 @@
+package ru.nsu.fit.oop.melnikov.game.snake.model.field.cell.objects;
+
+public interface CellObject {
```
Javadocs, логика работы коллизий с учетом приоритета неочевидна

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/model/field/cell/objects/EmptyCell.java:
```diff
@@ -0,0 +1,13 @@
+
+  @Override
+  public void onAnotherCellObjectAppearance(CellObject anotherCellObject) {
+    // Makes no effect
```
Это в Javadoc оверрайда

molo4evan commented on Task_2_3_1/src/main/java/ru/nsu/fit/oop/melnikov/game/snake/model/field/Field.java:
```diff
@@ -0,0 +1,103 @@
+    appleCells.add(newAppleCell);
+  }
+
+  public void eatApple(Cell appleCell) {
```
searchForApple?

andrey-dru-me1 committed:
make Cell extends Point so cell.getPoint.x() converts to cell.getX();
modify config file so there is no need to specify field size or make field square.

andrey-dru-me1 committed:
replace java.util timer with javafx timeline;
remove fieldPresenterTest because timeline doesn't work in non-javafx application thread, so it's impossible to test.

andrey-dru-me1 committed:
remove redundant field

andrey-dru-me1 committed:
convert grid to canvas. Not always correctly ends the game

andrey-dru-me1 committed:
fix finish screen

andrey-dru-me1 committed:
add spaces between cells

andrey-dru-me1 committed:
replace hardcoded colors with extendable set of textures

andrey-dru-me1 committed:
update jacoco version in task_1_3_1

andrey-dru-me1 committed:
fix 1_3_1 test

andrey-dru-me1 committed:
make creation of new texture packs easier

andrey-dru-me1 committed:
add animation;
turn snakeNode into record;
modify module-info and dependencies

andrey-dru-me1 committed:
remake rect class

andrey-dru-me1 committed:
add possibility to make animations. There is a bug with animations outside the cell's rect

andrey-dru-me1 committed:
add javadoc to draw method

andrey-dru-me1 committed:
refactor a little

andrey-dru-me1 committed:
make all the canvas be redrawn on each game tick

andrey-dru-me1 committed:
replace canvas.getContext2D() with context

andrey-dru-me1 committed:
rename stopThreads() to stopAnimations()

andrey-dru-me1 committed:
get rid of animations

andrey-dru-me1 committed:
remove the opportunity of turning snake against its previous movement

andrey-dru-me1 committed:
add possibility to move snake from one screen edge to another

andrey-dru-me1 committed:
add snake init direction in config file

andrey-dru-me1 committed:
add score

andrey-dru-me1 committed:
resize canvas on window resizing

andrey-dru-me1 committed:
add canvas resizing on window resizing

andrey-dru-me1 committed:
fix scene size changing;
make single repository of cell object DTOs' instead of one on each cell

andrey-dru-me1 committed:
fix scene size changing

andrey-dru-me1 committed:
decrease resize calls count

andrey-dru-me1 committed:
Simple stupid implementation of game end screen

andrey-dru-me1 committed:
improve end screen design

andrey-dru-me1 committed:
add settings screen (doesn't work yet);
add queue of user actions in presenter

andrey-dru-me1 committed:
make speed change work

andrey-dru-me1 committed:
remove possibility of changing snake direction on settings menu

andrey-dru-me1 committed:
add snake head

andrey-dru-me1 committed:
fix direction queue a little

andrey-dru-me1 committed:
add snake head rotation

andrey-dru-me1 committed:
add new screens

andrey-dru-me1 committed:
add map selection

andrey-dru-me1 committed:
improve settings screen

andrey-dru-me1 committed:
start implementing change keys screen

andrey-dru-me1 committed:
add back button to the change keys screen

andrey-dru-me1 committed:
implement key change key functionality

andrey-dru-me1 committed:
add bound keys displaying

andrey-dru-me1 committed:
improve game end screen

andrey-dru-me1 committed:
add scenes to fxmls

andrey-dru-me1 committed:
fix play again button

andrey-dru-me1 committed:
modify game pause screen and reformat all the code

andrey-dru-me1 committed:
make snake awaits for the first user's key pressing

andrey-dru-me1 committed:
fix bug

andrey-dru-me1 committed:
fix stage closing bug

andrey-dru-me1 committed:
make game settings be saved in file system

andrey-dru-me1 committed:
fix slider in settings screen

andrey-dru-me1 committed:
fix data loader;
add big empty map;
modify snake speed slider;
start implementing terminal snake

andrey-dru-me1 committed:
replace IntPoint with Point<Integer>

andrey-dru-me1 committed:
implement terminal snake

andrey-dru-me1 committed:
make GameSettings singleton

andrey-dru-me1 committed:
change a little

andrey-dru-me1 committed:
make a little refactoring

andrey-dru-me1 committed:
move terminal views of cell objects to separate classes

andrey-dru-me1 committed:
add possibility to select texture pack

andrey-dru-me1 committed:
add updating textures right while the game

andrey-dru-me1 committed:
add option for apple count

andrey-dru-me1 committed:
modify speed slider

andrey-dru-me1 committed:
add some javadocs

andrey-dru-me1 committed:
add model javadocs

andrey-dru-me1 committed:
add some terminal javadocs

andrey-dru-me1 committed:
add more javadocs and reformat code

andrey-dru-me1 committed:
divide increasing size and score into separate methods

andrey-dru-me1 committed:
make interfaces for snake

andrey-dru-me1 committed:
add use of destroyable interface

andrey-dru-me1 committed:
replace destroyable with snake because of bug

andrey-dru-me1 committed:
add snake body

andrey-dru-me1 committed:
Replace Point<Integer> with SnakePoint

andrey-dru-me1 committed:
move some methods from Snake to StaticSnake

andrey-dru-me1 committed:
move some methods from Snake to MovableSnake

andrey-dru-me1 committed:
small changes

andrey-dru-me1 committed:
change settings menu a little

andrey-dru-me1 committed:
start implementing GameData

andrey-dru-me1 committed:
wrap snake and field in gameData

andrey-dru-me1 committed:
delegate calculateNextPoint method to a field class

andrey-dru-me1 committed:
implement the simplest bot

andrey-dru-me1 committed:
make him a little smarter

andrey-dru-me1 committed:
make smart bot

andrey-dru-me1 committed:
make snake be destroyed if it moves to another snake;
fix bug with snake destroy

andrey-dru-me1 committed:
make bot extra-smart

andrey-dru-me1 committed:
modify config files and its loading so there is possibility to add as many bots as you want now

andrey-dru-me1 committed:
add generator
